[16.0][REL] l10n_do_accounting: v16 release

jose-pcg opened the pull request.

jose-pcg committed:
[WIP] l10n_do_accounting: v16 migration

jose-pcg committed:
[FIX] l10n_do_accounting: analytic accounts removed when invoice canceled (#1012)

jose-pcg committed:
[FIX] l10n_do_accounting: wrong document type behavior (#854) (#1019)

jose-pcg committed:
[FIX] l10n_do_accounting: wrong amount on qr stamp (#1020) (#1022)

jose-pcg committed:
[14.0][REF] l10n_do_accounting: test company address (#1023)

* [REF] l10n_do_accounting: test company address

* [REF] l10n_do_accounting: missing 'testing' context

jose-pcg committed:
[REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1024)

jose-pcg committed:
[REF] l10n_do_accounting: change the way to get ECF env for QR stamp

jose-pcg committed:
[IMP] l10n_do_accounting: eval context key to use resequence feature (#1027)

jose-pcg committed:
[IMP] l10n_do_accounting: improve l10n_do amounts calc (#1030)

jose-pcg committed:
[FIX] l10n_do_accounting: get amount total in company currency

jose-pcg committed:
[FIX] l10n_do_accounting: split company and invoice total amounts (#1033)

daniel-pcg committed:
[14.0][IMP] l10n_do_accounting: improvement in name_search (#1035)

* [IMP]l10n_do_accounting: improvement in name_search

* black solution

* black solution

crbll0 reviewed (approved):
Probado el flujo de facturación, todo correcto.

jose-pcg committed:
[REF] workflows: update actions

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[REF] l10n_do_accounting: using new _rec_names_search class attribute

jose-pcg committed:
[REF] l10n_do_accounting: use v16 flush and index

jose-pcg committed:
[REM] l10n_do_pos: remove deprecated module

jose-pcg committed:
[REM] l10n_do_pos: remove deprecated module

jose-pcg committed:
[REF] l10n_do_accounting: use logger for debbuging

jose-pcg committed:
[REF] l10n_do_accounting: try not installing

jose-pcg committed:
[REF] workflows: do not use postgres alpine

jose-pcg committed:
[15.0] *: merge 14.0 into 15.0 (#1044)

* [FIX] actions: remove uneeded dependencies

* [REF] README: update

* [REF] l10n_do_accounting: remove code

* [FIX] l10n_do_accounting: missing _get_last_sequence() param

* [FIX] l10n_do_accounting: added test tags

* [REM] remove deprecated module l10n_do_purchase

* [FIX] l10n_do_accounting: prevent KeyError calling l10n_do_dgii_tax_payer_type from account.journal (#966)

* [REF] workflows: update (#970)

* [IMP] l10n_do_accounting: l10n_do amounts function (#971)

* [IMP] l10n_do_accounting: implemented company_id field to journal Journal Document Type model (#975)

* [REM] requirements.txt

* [14.0][REF] workflows: actions update (#982) (#983)

* [REF] workflows: actions update (#982)

* [FIX] l10n_do_accounting: prevent KeyError if not active_model

* [REF] actions: artifact to build & push to GCR

Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>

* [REF] workflows: actions update

* [IMP] l10n_do_accounting: improve journal document types security (#984)

* [REF] l10n_do_accounting: unitest ref (#985)

* [REF] l10n_do_accounting: unitest ref (#986)

* [13.0][IMP] README: added repo readme (#977)

* [WIP] README: new readme

* [WIP] README: update

* [REF] README: update

* [REF] README: update

* [14.0][IMP] l10n_do_accounting: readme update (#990)

* [13.0][IMP] l10n_do_accounting: new readme (#978)

* [WIP] l10n_do_accounting: new readme

* [REF] l10n_do_accounting: readme typo

* [ADD] l10n_do_accounting: added company setup image

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* Update README.md

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: missing sequence_prefix & sequence_number calc on migration script (#989)

* [REF] workflows: update actions

* [FIX] l10n_do_accounting: prevent dependency chain to be broken on button_cancel() (#1004)

* [FIX] l10n_do_accounting: analytic accounts removed when invoice canceled (#1011)

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1016)

* [REF] l10n_do_accounting: eval context key on fiscal sequence fetching (#1017)

* [FIX] l10n_do_accounting: wrong document type behavior (#854) (#1019)

* [FIX] l10n_do_accounting: wrong amount on qr stamp (#1020) (#1021)

* [14.0][REF] l10n_do_accounting: test company address (#1023)

* [REF] l10n_do_accounting: test company address

* [REF] l10n_do_accounting: missing 'testing' context

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1024)

* [REF] l10n_do_accounting: change the way to get ECF env for QR stamp

* [IMP] l10n_do_accounting: eval context key to use resequence feature (#1026)

* [IMP] l10n_do_accounting: improve l10n_do amounts calc (#1029)

* [FIX] l10n_do_accounting: get amount total in company currency

* [FIX] l10n_do_accounting: split company and invoice total amounts (#1032)

* [14.0][IMP] l10n_do_accounting: improvement in name_search (#1035)

* [IMP]l10n_do_accounting: improvement in name_search

* black solution

* black solution

* [REM] l10n_do_pos: remove deprecated module

* [FIX] l10n_do_accounting: prevent MemoryError when loading database with a lot of moves

* [FIX] l10n_do_accounting: 2nd attempt to prevent MemoryError when loading database

Prevent MemoryError when loading database with lots of moves.

* [FIX] l10n_do_accounting: not all records computed

* [FIX] l10n_do_accounting: 4th attempt to prevent MemoryError when loading database

 Prevent MemoryError when loading database with lots of moves.

* [IMP] l10n_do_accounting: improve memory usage when load l10n_latam.document.type.csv

* [FIX] l10n_do_accounting: wrong fields on depends

* [REF] l10n_do_accounting: assert after post()

---------

Co-authored-by: Gustavo Valverde <gustavo@iterativo.do>
Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>
Co-authored-by: Daniel Alexander Pereyra Beltran <95594662+DanielAPereyraB@users.noreply.github.com>

jose-pcg committed:
[FIX] l10n_do_accounting: prevent url sign date quoting twice (#1049)

jose-pcg committed:
[IMP] l10n_do_accounting: improve error message (#1050)

jose-pcg committed:
[REF] l10n_do_accounting: update unitest (#1051)

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[REF] l10n_do_accounting: ref test

jose-pcg committed:
[REF] l10n_do_accounting: lol

jose-pcg committed:
[FIX] l10n_do_accounting: use model_create_multi

jose-pcg committed:
[15.0][FWP] l10n_do_accounting: merge 14.0 into 15.0 (#1055)

* [FIX] actions: remove uneeded dependencies

* [REF] README: update

* [REF] l10n_do_accounting: remove code

* [FIX] l10n_do_accounting: missing _get_last_sequence() param

* [FIX] l10n_do_accounting: added test tags

* [REM] remove deprecated module l10n_do_purchase

* [FIX] l10n_do_accounting: prevent KeyError calling l10n_do_dgii_tax_payer_type from account.journal (#966)

* [REF] workflows: update (#970)

* [IMP] l10n_do_accounting: l10n_do amounts function (#971)

* [IMP] l10n_do_accounting: implemented company_id field to journal Journal Document Type model (#975)

* [REM] requirements.txt

* [14.0][REF] workflows: actions update (#982) (#983)

* [REF] workflows: actions update (#982)

* [FIX] l10n_do_accounting: prevent KeyError if not active_model

* [REF] actions: artifact to build & push to GCR

Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>

* [REF] workflows: actions update

* [IMP] l10n_do_accounting: improve journal document types security (#984)

* [REF] l10n_do_accounting: unitest ref (#985)

* [REF] l10n_do_accounting: unitest ref (#986)

* [13.0][IMP] README: added repo readme (#977)

* [WIP] README: new readme

* [WIP] README: update

* [REF] README: update

* [REF] README: update

* [14.0][IMP] l10n_do_accounting: readme update (#990)

* [13.0][IMP] l10n_do_accounting: new readme (#978)

* [WIP] l10n_do_accounting: new readme

* [REF] l10n_do_accounting: readme typo

* [ADD] l10n_do_accounting: added company setup image

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* Update README.md

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: missing sequence_prefix & sequence_number calc on migration script (#989)

* [REF] workflows: update actions

* [FIX] l10n_do_accounting: prevent dependency chain to be broken on button_cancel() (#1004)

* [FIX] l10n_do_accounting: analytic accounts removed when invoice canceled (#1011)

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1016)

* [REF] l10n_do_accounting: eval context key on fiscal sequence fetching (#1017)

* [FIX] l10n_do_accounting: wrong document type behavior (#854) (#1019)

* [FIX] l10n_do_accounting: wrong amount on qr stamp (#1020) (#1021)

* [14.0][REF] l10n_do_accounting: test company address (#1023)

* [REF] l10n_do_accounting: test company address

* [REF] l10n_do_accounting: missing 'testing' context

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1024)

* [REF] l10n_do_accounting: change the way to get ECF env for QR stamp

* [IMP] l10n_do_accounting: eval context key to use resequence feature (#1026)

* [IMP] l10n_do_accounting: improve l10n_do amounts calc (#1029)

* [FIX] l10n_do_accounting: get amount total in company currency

* [FIX] l10n_do_accounting: split company and invoice total amounts (#1032)

* [14.0][IMP] l10n_do_accounting: improvement in name_search (#1035)

* [IMP]l10n_do_accounting: improvement in name_search

* black solution

* black solution

* [REM] l10n_do_pos: remove deprecated module

* [14.0] *: fwp 13.0 into 14.0 (#1043)

* [FIX] l10n_do_accounting: prevent MemoryError when loading database with a lot of moves

* [FIX] l10n_do_accounting: 2nd attempt to prevent MemoryError when loading database

Prevent MemoryError when loading database with lots of moves.

* [FIX] l10n_do_accounting: not all records computed

* [FIX] l10n_do_accounting: 4th attempt to prevent MemoryError when loading database

 Prevent MemoryError when loading database with lots of moves.

* [IMP] l10n_do_accounting: improve memory usage when load l10n_latam.document.type.csv

* [FIX] l10n_do_accounting: wrong fields on depends

* [REF] l10n_do_accounting: assert after post()

---------

Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>

* [FIX] l10n_do_accounting: prevent url sign date quoting twice (#1048)

* [IMP] l10n_do_accounting: improve error message (#1050)

* [14.0][FWP] l10n_do_accounting: merge 13.0 into 14.0 (#1054)

* [FWP] l10n_do_accounting: merge 13.0 into 14.0

* [FIX] l10n_do_accounting: electronic stamp test

---------

Co-authored-by: Gustavo Valverde <gustavo@iterativo.do>
Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>
Co-authored-by: Daniel Alexander Pereyra Beltran <95594662+DanielAPereyraB@users.noreply.github.com>

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[REF] l10n_do_accounting: don't use quote_plus (#1056)

jose-pcg committed:
[REF] l10n_do_accounting: don't use quote_plus (#1056)

jose-pcg committed:
[REF] l10n_do_accounting: improve tax data functions

jose-pcg committed:
[FIX] l10n_do_accounting: remove extra character form qr stamp url (#1058) (#1060)

jose-pcg committed:
[FIX] l10n_do_accounting: remove extra character form qr stamp url (#1058) (#1060)

jose-pcg committed:
[14.0][FIX] l10n_do_accounting: fix base_amount and l10n_do_invoice_total calc (#1063) (#1064)

* [FIX] l10n_do_accounting: fix base_amount and l10n_do_invoice_total calc (#1061)

* [REF] l10n_do_accounting: update module version (#1062)

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[FIX] l10n_do_accounting: wrong exempt amount in report (#1066)

jose-pcg committed:
[FIX] l10n_do_accounting: wrong exempt amount in report (#1066)

jose-pcg committed:
[15.0] *: Merge branch '14.0' into 15.0-fix-002-jl (#1069)

* [FIX] actions: remove uneeded dependencies

* [REF] README: update

* [REF] l10n_do_accounting: remove code

* [FIX] l10n_do_accounting: missing _get_last_sequence() param

* [FIX] l10n_do_accounting: added test tags

* [REM] remove deprecated module l10n_do_purchase

* [FIX] l10n_do_accounting: prevent KeyError calling l10n_do_dgii_tax_payer_type from account.journal (#966)

* [REF] workflows: update (#970)

* [IMP] l10n_do_accounting: l10n_do amounts function (#971)

* [IMP] l10n_do_accounting: implemented company_id field to journal Journal Document Type model (#975)

* [REM] requirements.txt

* [14.0][REF] workflows: actions update (#982) (#983)

* [REF] workflows: actions update (#982)

* [FIX] l10n_do_accounting: prevent KeyError if not active_model

* [REF] actions: artifact to build & push to GCR

Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>

* [REF] workflows: actions update

* [IMP] l10n_do_accounting: improve journal document types security (#984)

* [REF] l10n_do_accounting: unitest ref (#985)

* [REF] l10n_do_accounting: unitest ref (#986)

* [13.0][IMP] README: added repo readme (#977)

* [WIP] README: new readme

* [WIP] README: update

* [REF] README: update

* [REF] README: update

* [14.0][IMP] l10n_do_accounting: readme update (#990)

* [13.0][IMP] l10n_do_accounting: new readme (#978)

* [WIP] l10n_do_accounting: new readme

* [REF] l10n_do_accounting: readme typo

* [ADD] l10n_do_accounting: added company setup image

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* Update README.md

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [WIP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: readme update

* [IMP] l10n_do_accounting: missing sequence_prefix & sequence_number calc on migration script (#989)

* [REF] workflows: update actions

* [FIX] l10n_do_accounting: prevent dependency chain to be broken on button_cancel() (#1004)

* [FIX] l10n_do_accounting: analytic accounts removed when invoice canceled (#1011)

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1016)

* [REF] l10n_do_accounting: eval context key on fiscal sequence fetching (#1017)

* [FIX] l10n_do_accounting: wrong document type behavior (#854) (#1019)

* [FIX] l10n_do_accounting: wrong amount on qr stamp (#1020) (#1021)

* [14.0][REF] l10n_do_accounting: test company address (#1023)

* [REF] l10n_do_accounting: test company address

* [REF] l10n_do_accounting: missing 'testing' context

* [REF] l10n_do_accounting: send 'testing' key through context to prevent ecf sending (#1024)

* [REF] l10n_do_accounting: change the way to get ECF env for QR stamp

* [IMP] l10n_do_accounting: eval context key to use resequence feature (#1026)

* [IMP] l10n_do_accounting: improve l10n_do amounts calc (#1029)

* [FIX] l10n_do_accounting: get amount total in company currency

* [FIX] l10n_do_accounting: split company and invoice total amounts (#1032)

* [14.0][IMP] l10n_do_accounting: improvement in name_search (#1035)

* [IMP]l10n_do_accounting: improvement in name_search

* black solution

* black solution

* [REM] l10n_do_pos: remove deprecated module

* [14.0] *: fwp 13.0 into 14.0 (#1043)

* [FIX] l10n_do_accounting: prevent MemoryError when loading database with a lot of moves

* [FIX] l10n_do_accounting: 2nd attempt to prevent MemoryError when loading database

Prevent MemoryError when loading database with lots of moves.

* [FIX] l10n_do_accounting: not all records computed

* [FIX] l10n_do_accounting: 4th attempt to prevent MemoryError when loading database

 Prevent MemoryError when loading database with lots of moves.

* [IMP] l10n_do_accounting: improve memory usage when load l10n_latam.document.type.csv

* [FIX] l10n_do_accounting: wrong fields on depends

* [REF] l10n_do_accounting: assert after post()

---------

Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>

* [FIX] l10n_do_accounting: prevent url sign date quoting twice (#1048)

* [IMP] l10n_do_accounting: improve error message (#1050)

* [14.0][FWP] l10n_do_accounting: merge 13.0 into 14.0 (#1054)

* [FWP] l10n_do_accounting: merge 13.0 into 14.0

* [FIX] l10n_do_accounting: electronic stamp test

* [REF] l10n_do_accounting: don't use quote_plus (#1056)

* [FIX] l10n_do_accounting: remove extra character form qr stamp url (#1058)

* [14.0][FIX] l10n_do_accounting: fix base_amount and l10n_do_invoice_total calc (#1063)

* [FIX] l10n_do_accounting: fix base_amount and l10n_do_invoice_total calc (#1061)

* [REF] l10n_do_accounting: update module version (#1062)

* [FIX] l10n_do_accounting: wrong exempt amount in report (#1066)

* [FIX] l10n_do_accounting: consider 1.8 and 1.6 ITBIS tax amounts (#1067) (#1068)

---------

Co-authored-by: Gustavo Valverde <gustavo@iterativo.do>
Co-authored-by: Andres Rodriguez <androdriguez14@gmail.com>
Co-authored-by: Daniel Alexander Pereyra Beltran <95594662+DanielAPereyraB@users.noreply.github.com>

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[IMP] l10n_do_accounting: set '%' as a safe character for DGII URL compose (#1070) (#1072)

jose-pcg committed:
Merge branch '15.0' into 16.0-mig-001-jl

jose-pcg committed:
[WIP] l10n_do_accounting: working with itbis 18 tax amount

jose-pcg committed:
[IMP] l10n_do_accounting: set Gasto Menor no matter company partner tax payer type (#1074)

jose-pcg committed:
[IMP] l10n_do_accounting: load report_templates.xml data

jose-pcg committed:
[IMP] l10n_do_accounting: debit notes now works ok

jose-pcg committed:
[IMP] l10n_do_accounting: credit notes now works ok

Commit:
[WIP] l10n_do_accounting: working with unitest

Commit:
[WIP] l10n_do_accounting: working with unitest

jose-pcg committed:
[WIP] l10n_do_accounting: working with unitest 1
